frontend/src/views/files/Listing.vue

```diff
@@ -394,6 +394,11 @@ export default {
 
           // Fill and fit the window with listing items
           this.fillWindow(true);
+          if (window.sessionStorage.getItem(this.$route.path)) {
+            this.addSelected(JSON.parse(window.sessionStorage.getItem(this.$route.path))[1]);
+            document.documentElement.scrollTop = JSON.parse(window.sessionStorage.getItem(this.$route.path))[0];
+	    window.sessionStorage.removeItem(this.$route.path);
+          }
```

Comment:
Here is the only good place to set the scrolltop and selected item to the saved values.